test/Microsoft.AspNet.Mvc.Core.Test/HttpResponseStreamWriterTest.cs

```diff
@@ -0,0 +1,337 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class HttpResponseStreamWriterTest
+    {
+        [Fact]
+        public async Task DoesNotWriteBOM()
```

Comment:
Tests look good to start with - but we should cover other Encodings as part of the writer.